miqa/core/models/evaluation.py

```diff
@@ -0,0 +1,25 @@
+from uuid import uuid4
+
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from miqa.learning.evaluation_models import available_evaluation_models
+
+
+class Evaluation(models.Model):
+    id = models.UUIDField(primary_key=True, default=uuid4, editable=False)
+    image = models.ForeignKey('Image', null=False, on_delete=models.CASCADE)
+    evaluation_model = models.CharField(max_length=50)
+    results = models.JSONField()
+
+    def __str__(self):
+        return f'Evaluation for {str(self.image.raw_path)}'
+
+    def save(self, *args, **kwargs):
+        evaluation_model = available_evaluation_models[self.evaluation_model]
+        if any([key not in evaluation_model.expected_outputs for key in self.results.keys()]):
+            raise ValidationError(
+                'Result keys must match expected outputs of associated evaluation model.'
+            )
```

Comment:
Let's move validation logic into `full_clean` rather than overriding `save`. Then we'll need to call that in places where we make the model (at least if we're using user-provided input to these fields).

Reply:
We actually don't use any user input when creating Evaluation objects. This is just a sanity check. Should we move it to `full_clean` still?

Reply:
Oh, in that case I would actually just delete this code. My general philosophy here is that we shouldn't include code to try to protect us against bugs of our own making, unless there's some compelling reason to believe we would cause a specific bug (which is maybe a design red flag that could be addressed with refactoring).

Reply:
Okay, removing the check. The only place we would create Evaluation objects is in the one place it is currently called, in `evaluate_images` in `tasks.py`.